src/main/kotlin/uk/gov/dluhc/printapi/service/CertificateBatchingService.kt

```diff
@@ -29,12 +35,40 @@ class CertificateBatchingService(
 
     fun batchCertificates(batchSize: Int): Map<String, List<Certificate>> {
         val certificatesPendingAssignment = certificateRepository.findByStatus(Status.PENDING_ASSIGNMENT_TO_BATCH)
-        return certificatesPendingAssignment.chunked(batchSize).associate { batchOfCertificates ->
+
+        return limitCertificates(certificatesPendingAssignment).chunked(batchSize).associate { batchOfCertificates ->
             val batchId = idFactory.batchId()
             batchId to batchOfCertificates.onEach {
                 it.getCurrentPrintRequest().batchId = batchId
                 it.addStatus(Status.ASSIGNED_TO_BATCH)
             }
         }
     }
+
+    private fun limitCertificates(certificatesPendingAssignment: List<Certificate>): List<Certificate> {
+        val startOfDay = Instant.now(clock).truncatedTo(DAYS)
+        val endOfDay = startOfDay.plus(1, DAYS).minusSeconds(1)
+        val countOfRequestsSentToPrintProvider =
+            certificateRepository.getPrintRequestStatusCount(startOfDay, endOfDay, Status.ASSIGNED_TO_BATCH)
+
+        if (certificatesPendingAssignment.size + countOfRequestsSentToPrintProvider > dailyLimit) {
```

Comment:
please can brackets be added so don't have to worry about the precedence order of the different operators